### PR TITLE
Better Error Messages

### PR DESCRIPTION
**Pull request by Nathan-Fenner**
Replace go-peg's errors with useful ones. This requires a lot of additional annotation, and there's still probably some uncaught errors.

@drcapulet

### CODE REVIEW

File: query/parser/language.peg [hunk 1]
```diff
@@ -50,31 +54,50 @@ root <- (selectStmt / describeStmt) _ !.
...
   { p.addEvaluationContext() }
-  ( _ PROPERTY_KEY   { p.addPropertyKey(buffer[begin:end])   }
-    _ PROPERTY_VALUE { p.addPropertyValue(buffer[begin:end]) }
+  (
+    _ PROPERTY_KEY { p.addPropertyKey(text) }
+    (
+      _ PROPERTY_VALUE {
+      p.addPropertyValue(text) }
+      /
+      &{ p.errorHere(position, `expected value to follow key '%s'`, p.contents(tree, tokenIndex-2)   ) }
```
Comment (drcapulet) [inline query/parser/language.peg]: Nit extra spaces between `p.contents(tree, tokenIndex-2)` and `) }`

File: query/parser/language.peg [hunk 2]
```diff
@@ -127,25 +159,29 @@ add_one_pipe <-
...
-  _ STRING { p.addStringNode(unescapeLiteral(buffer[begin:end])) }
+  _ <NUMBER> { p.addNumberNode(text) } /
+  _ STRING { p.addStringNode(unescapeLiteral(text)) }
 
 expression_annotation_required <-
   _ "{"
   <[^}]*>
-  "}"
-  { p.addAnnotationExpression(buffer[begin:end]) }
+  ("}" / &{ p.errorHere(position, `expected "%CLOSEBRACE%" to close "%OPENBRACE%" opened for annotation`) })
```
Comment (drcapulet) [inline query/parser/language.peg]: Why the `"%CLOSEBRACE%"` and `"%OPENBRACE%"`? Those special?

File: query/parser/language.peg (hunk 1 above)
Comment (Nathan-Fenner) [inline query/parser/language.peg]: Fixed

File: query/parser/language.peg (hunk 2 above)
Comment (Nathan-Fenner) [inline query/parser/language.peg]: Oh- I actually need to handle these still.

The PEG parser generator doesn't actually handle Go's grammar in `{ }` blocks; it just assumes that they'll match up and when the last one closes, the block does too. This means they can't go in strings or comments without confusing go-peg a little.

File: testing_support/script/verify-build [hunk 3]
```diff
@@ -48,6 +48,10 @@ if [ "$before" != "$after" ]; then
 	echo "Make sure you ran the build file, and that your version of peg is up to date."
 	echo "To get the latest version of peg, run:"
 	echo "> go get -u github.com/pointlander/peg"
+	echo "expected:"
+	echo "======================================================================="
+	echo $(cat ./query/parser/language.peg.go)
```
Comment (drcapulet) [inline testing_support/script/verify-build]: Any reason we want to output this? Considering the file is a few thousand lines.

Comment (Nathan-Fenner) [inline testing_support/script/verify-build]: I used it for debugging an issue (that turned out to not really be related to this), but it's best to remove it.

**Comment by drcapulet** [comment]
LGTM, mind squashing?

**Comment by drcapulet** [comment]
Still LGTM

**Pull request merged**